web/text_layer_builder.js

```diff
@@ -184,25 +192,68 @@ var TextLayerBuilder = (function TextLayerBuilderClosure() {
       if (angle !== 0) {
         textDiv.dataset.angle = angle * (180 / Math.PI);
       }
-      // We don't bother scaling single-char text divs, because it has very
-      // little effect on text highlighting. This makes scrolling on docs with
-      // lots of such divs a lot faster.
-      if (textDiv.textContent.length > 1) {
-        if (style.vertical) {
-          textDiv.dataset.canvasWidth = geom.height * this.viewport.scale;
-        } else {
-          textDiv.dataset.canvasWidth = geom.width * this.viewport.scale;
-        }
-      }
+      return textDiv;
     },
 
     setTextContent: function TextLayerBuilder_setTextContent(textContent) {
+      // This function will add the text divs and append them to the DOM
       this.textContent = textContent;
 
       var textItems = textContent.items;
-      for (var i = 0, len = textItems.length; i < len; i++) {
-        this.appendText(textItems[i], textContent.styles);
+      var len = textItems.length;
+      
+      var textDivs = []; // Just temporary
+      for (var i = 0; i < len; i++) {
+        textDivs.push(this.appendText(textItems[i], textContent.styles));
+      }
+      
+      // Set each element's padding to run to the nearest right and bottom 
+      // element. The padding ensures that text selection works.
+      var page_w = this.textLayerDiv.offsetWidth;
+      var page_h = this.textLayerDiv.offsetHeight;
+      var scale  = this.viewport.scale;
+      for (i = 0; i < len; i++) {
+        var geom = textItems[i];
+        var divi = textDivs[i];
+        
+        if(geom.div_angle) {
+          // Angled text is more complex and outside the scope... for now.
+          continue;
+        }
+        
+        var bottom  = geom.div_top + geom.height*scale;
+        var right   = geom.div_left + geom.width*scale;
+        
+        var far_right = geom.right !== null ?
+                          textItems[geom.right].div_left : page_w;
+        var far_bottom = geom.bottom !== null ?
+                          textItems[geom.bottom].div_top : page_h;
+// #if !PRODUCTION
```

Comment:
I'm not sure if this should be in the codebase if it's only for debugging purposes.